ssg/build_cpe.py

```diff
@@ -189,72 +205,109 @@ def to_xml_element(self):
         super(CPEALFactRef, self).__init__(obj)
         self.cpe_name = obj  # we do not want to modify original name used for platforms
-        self.bash_conditional = ""
-        self.ansible_conditional = ""
-
-    def enrich_with_cpe_info(self, cpe_products):
-        self.bash_conditional = cpe_products.get_cpe(self.cpe_name).bash_conditional
-        self.ansible_conditional = cpe_products.get_cpe(self.cpe_name).ansible_conditional
-        self.cpe_name = cpe_products.get_cpe_name(self.cpe_name)
+        self.conditional = {}
+
+    def add_enriched_cpe_items(self, product_cpes):
+        old_cpe_dict = product_cpes.get_cpe(self.cpe_name).represent_as_dict()
+        # ignore remediation snippets for now when templating
+        # they will be removed eventually from the CPE item definition
+        not_templated_keys = ["conditional"]
+        new_cpe_dict = apply_formatting_on_dict_values(
+            old_cpe_dict, self.as_dict(), not_templated_keys)
+        new_cpe_dict["id_"] = self.as_id()
+        new_cpe = CPEItem.get_instance_from_full_dict(new_cpe_dict)
+        self.cpe_name = product_cpes.get_cpe_name(self.cpe_name)
+        product_cpes.add_cpe_item(new_cpe)
 
     def to_xml_element(self):
         cpe_factref = ET.Element("{%s}fact-ref" % CPEALFactRef.ns)
-        cpe_factref.set('name', self.cpe_name)
+        cpe_factref.set('name', self.cpe_name.format(**self.as_dict()))
 
         return cpe_factref
 
-    def to_bash_conditional(self):
-        return self.bash_conditional
+    def get_bash_conditional(self, product_cpes, conditionals_path, env_yaml):
+        cpe = product_cpes.get_cpe(self.as_id())
+        cond = cpe.conditional.get("bash", "")
+        if cond:
+            return remediations.parse_from_string_with_jinja(cond, env_yaml)
+        elif conditionals_path is not None:
+            templated_conditional_file = os.path.join(conditionals_path, 'bash', self.as_id() +
+                                                      remediations.REMEDIATION_TO_EXT_MAP['bash'])
+            if os.path.exists(templated_conditional_file):
+                return remediations.parse_from_file_without_jinja(templated_conditional_file)
+        return None
+
+    def get_ansible_conditional(self, product_cpes, conditionals_path, env_yaml):
+        cpe = product_cpes.get_cpe(self.as_id())
+        cond = cpe.conditional.get("ansible", "")
+        if cond:
+            return remediations.parse_from_string_with_jinja(cond, env_yaml)
+        elif conditionals_path is not None:
+            templated_conditional_file = os.path.join(conditionals_path, "ansible", self.as_id() +
+                                                      remediations.REMEDIATION_TO_EXT_MAP[
+                                                          'ansible'])
+            if os.path.exists(templated_conditional_file):
+                return remediations.parse_from_file_without_jinja(templated_conditional_file)
+        return None
```

Comment:
These 2 methods look very similar. Would it make sense to extract the common code in a single method?

Reply:
I tried to deduplicate the code here:
https://github.com/ComplianceAsCode/content/pull/8923/commits/e87430f98860d87dd396988c179f9967ab0ec42e